hack/check_go_version.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+first_version=""
+# Find and iterate over all go.mod files, excluding the vendor directory
+find . -name go.mod | while read -r mod; do
+    current_version=$(grep 'go 1.' "$mod" | cut -d ' ' -f 2)
```

Comment:
```suggestion
    current_version=$(grep '^go 1\.' "$mod" | cut -d ' ' -f 2)
```

a little stronger regex, not that this really needs to be bulletproof. Using `.` w/o `\.` means match any single character. `^` ensures our expression _starts_ a line.